hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -967,6 +1083,21 @@ case class MatrixVCFReader(
     rowKey = Array("locus", "alleles"),
     entryType = genotypeSignature)
 
+  val partitioner = if (partitionsJSON != null) {
+    assert(inputs.length == 1)
+    assert(minPartitions.isEmpty)
+    assert(hConf.exists(inputs.head + ".tbi"))
+
+    val pkType = TArray(TInterval(TStruct("locus" -> locusType)))
+    val jv = JsonMethods.parse(partitionsJSON)
+    val rangeBounds = JSONAnnotationImpex.importAnnotation(jv, pkType)
+
+    new RVDPartitioner(fullType.rowKeyStruct,
+      rangeBounds.asInstanceOf[IndexedSeq[Interval]],
+      0)
```

Comment:
Since the meaning of `allowedOverlap` is a bit tricky, it might be clearer to readers to express this using the constructor that takes a `partitionKey`:
```scala
new RVDPartitioner(
  Array("locus"),
  fullType.rowKeyStruct,
  rangeBounds.asInstanceOf[IndexedSeq[Interval]])
```
Also, I assume you're sure no locus can appear in multiple partitions?

Reply:
I'm fairly sure that if loci are duplicated and out of order something is terribly wrong. If they are in order, then the index should just grab all of them.